File: classdemojava/src/main/java/com/example/classdemojava/classDemo.java
package com.example.classdemojava;

/**
 * Created by LENOVO on 23/03/2018.
 */
//Simple class
class c{
    void fun(){
        System.out.println("This  is a simple class");
    }
}

//Constructor


class c1{
    //Default Constructor
    c1(){
        System.out.println("This is default constructor");
    }
    //parameterised constructor
    c1(String str){
        System.out.println("This is parameterised constructor");
        System.out.println("Value of string is:"+str);
    }
    void Display(){
        System.out.println("This is display method in class c1");
    }
}

//inheritance
class c2 extends c1{
    void  show(){
        System.out.println("this is show method in c2 class");
        super.Display();
    }
    //overiding
    void Display(){
        System.out.println("This is display method in class c2");
    }
}

//Inteface
interface i1{
    int cal(int a,int b);
}
class c3 implements i1{
    int c=10;
    @Override
    public int cal(int a, int b) {
        return a+b+c;
    }
    public int multi(int no){
        return c*no;
    }
}
class c4 extends c3 implements i1{
     public int cal(int a,int b){
         return a*b;
     }
}

public class classDemo {
    public static void main(String args[]){
            c ob=new c();
            ob.fun();

            c1 ob1=new c1();
            c1 ob2=new c1("hello");
            ob1.Display();

            c2 ob3=new c2();
            ob3.show();
            ob3.Display();

            c3 ob4=new c3();
            ob4.cal(15,25);
            ob4.multi(30);

            c4 ob5=new c4();
            ob5.cal(50,60);
            ob5.multi(12);


    }
}

File: app/src/main/java/com/example/lenovo/classdemo/classDemo.kt
package com.example.lenovo.classdemo

/**
 * Created by LENOVO on 13/03/2018.
 */

//simple class
class A{
    fun m1()
    {
        println("this is simple class")
    }
}

//primary constructor
class B(private var no:Int){
    fun m2(){
        println("Value of no. is $no")
    }
}

//secondary constructor
class C{
    private var name:String="Hello"
    constructor(){
        this.name
        //println("$name")
    }
    fun m(){
        println("$name")
    }
}

open class C1{
    private var name:String="Hi"
    constructor(str:String){
        this.name=str
    }
    fun m(){
        println("$name")
    }
}
//inheritance

open class Child(private var str:String="hi",private var str1:String ): C1(str) {
    fun display(){
        println("$str $str1")
    }
}

//interface
interface Intr1{
    fun show()
}

open class Child2:Intr1{

    override fun show(){
        println("It is an example of implementing interface in child2 class")
    }
}

class Child3(private var str:String,private var str1:String ):Child(str,str1),Intr1{
    override fun show() {
        println("It is an example of implementing interface in child3 class")
        display()
        m()
    }

    fun fun1(vararg string: String){
        for(str in string)
        {
            println(str)
        }
    }
}


fun main(args: Array<String>) {
    var a=A()
    a.m1()

    val b=B(10)
    b.m2()

    val c=C()
    c.m()

    val c1=C1("Hello")
    c1.m()

    val ob=Child("hello",str1 = "Ravina")
    ob.display()
    ob.m()

    val ob1=Child(str1="Ravina")
    ob1.display()

    val obj=Child2()
    obj.show()
    //or
    val obj1: Child2=object : Child2() {
        override fun show() {
            println("Interface method implementation")
        }
    }

    val obj2=Child3("hello",str1 = "friends")
    obj2.show()

    obj2.fun1("one")
    obj2.fun1("one","two","three")
    obj2.fun1()
}